src/node_messaging.cc

```diff
@@ -1136,14 +1136,42 @@ void JSTransferable::New(const FunctionCallbackInfo<Value>& args) {
   new JSTransferable(Environment::GetCurrent(args), args.This());
 }
 
+Local<Function> JSTransferable::GetConstructorFunction(Isolate* isolate) {
+  Local<FunctionTemplate> js_transferable_template = FunctionTemplate::New(
+      isolate,
+      New,
+      Local<Value>(),
+      Local<v8::Signature>(),
+      0,
+      v8::ConstructorBehavior::kAllow,
+      v8::SideEffectType::kHasSideEffect,
+      nullptr);
+  Local<FunctionTemplate> tmpl = FunctionTemplate::New(
```

Comment:
Judging from the sudden rise in the number of DataCloneErrors from https://github.com/nodejs/node/pull/41148/commits/529143809e9241896bded1d64fd597bf2eb33121 (which uses the JSTransferable constructor factory function only for the per_context modules) to https://github.com/nodejs/node/pull/41148/commits/9353cafcdf9d5d8314e50af2d505473c14cd04f4 (which uses the same factory for both the context-aware/unaware modules), I think the reason why this doesn't work yet is that we need to use the same BaseObject constructor template that is set only once in https://github.com/nodejs/node/blob/ced5fc39303438f09f09c3036afc69456ee552bd/src/env.cc#L1712-L1720 because conditionals like https://github.com/nodejs/node/blob/c8f5dd64b157280297fbea7bbef6cca66b587c27/src/node_messaging.cc#L291 might not work because these are technically not the same objects. I guess it's necessary to use `Isolate::Get/SetData` to create and store the template before the Environment is created inside something like https://github.com/nodejs/node/blob/ced5fc39303438f09f09c3036afc69456ee552bd/src/api/environment.cc#L307-L312. Not sure if a better option exists.